lib/i18n/tasks/wrong_interpolations.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module I18n::Tasks
+  module WrongInterpolations
+    VARIABLE_REGEX = /%{[^}]+}/
+
+    def wrong_interpolations(locales: nil, base_locale: nil)
+      locales ||= self.locales
+      base      = base_locale || self.base_locale
+      tree      = empty_forest
+
+      data[base].key_values.each do |key, node|
+        next unless node.is_a?(String)
+
+        base_variables = node.scan(VARIABLE_REGEX)
+
+        (locales - [base]).each do |current_locale|
+          check_in_other_locale(tree, key, base_variables, current_locale)
+        end
+      end
+
+      tree
+    end
+
+    private
+
+    def check_in_other_locale(tree, key, base_variables, locale)
+      node = data[locale].first.children[key]
+
+      return if node.nil? || !node.value.is_a?(String)
+
+      compare_variables = node.value.scan(VARIABLE_REGEX)
+
+      return if same_array?(base_variables, compare_variables)
+
+      tree.merge!(select_key(key, locale))
+    end
+
+    def select_key(key, locale)
+      data[locale].select_keys(root: false) { |search_key| search_key == key }
```

Comment:
This is rather inefficient as it traverses the whole tree. Consider something like this instead:

```ruby
tree.append!(
  node.walk_to_root.reduce(nil) { |p, c| [p.derive(children: c)] }.tap do |root|
    root.data[:type] = :inconsistent_interpolation
  end
)
```

Reply:
That code does not work as is, and I do not understand walk_to_root/derive enough to figure out why. Won't append also fail when multiple nodes come from the same locale e.g. `es`?